nixos/modules/services/web-apps/healthchecks.nix

```diff
@@ -0,0 +1,216 @@
+{ config, lib, pkgs, buildEnv, ... }:
+
+with lib;
+
+let
+  cfg = config.services.healthchecks;
+  staticDir = cfg.dataDir + "/static";
+  boolToPython = b: if b then "True" else "False";
+  localSettingsFile = pkgs.writeTextFile {
+    name = "local_settings.py";
+    text = ''
+      STATIC_ROOT = '${staticDir}'
+      ALLOWED_HOSTS = ${builtins.toJSON cfg.allowedHosts}
+      # Default database engine is SQLite. So one can just check out code,
+      # install requirements.txt and do manage.py runserver and it works
+      DEFAULT_AUTO_FIELD = "django.db.models.AutoField"
+      DATABASES = {
+          "default": {
+          "ENGINE": "django.db.backends.sqlite3",
+          "NAME": "${cfg.dataDir}/healthchecks.sqlite",
+        }
+      }
+      with open("${cfg.secretKeyFile}", "r") as file:
+          SECRET_KEY = file.readline()
+      DEBUG = ${boolToPython cfg.debug}
+      REGISTRATION_OPEN = ${boolToPython cfg.registrationOpen}
+
+      ${cfg.extraConfig}
+    '';
+  };
+  pkg = (pkgs.healthchecks.overrideAttrs (old: {
+    installPhase = old.installPhase + ''
+      ln -s ${localSettingsFile} $out/opt/healthchecks/hc/local_settings.py
+    '';
```

Comment:
This would cause us to rebuild the package everytime. If we cannot give the config via an argument we should think about a wrapper.

Reply:
Hm.. this is how it's done in netbox atm, but that's a very valid concern. An alternative would be to patch settings.py to load local_settings from somewhere else, but for that we would need a path which is known at build time. Not sure what's the best way forward here, will investigate how other django apps in nixpkgs approach this problem.

Reply:
Another approach would be to create an extra directory in the nix store, containing both the default config and our changes, and put that in PYTHON_PATH before the default one, so effectively shadowing the one in the package. 
that would be similar to how nixos/modules/services/monitoring/graphite.nix does it. 

The thing is, we would need to patch BASE_DIR then in order to keep it working. Just patching settings.py to replace 
```
# Read additional configuration from hc/local_settings.py if it exists
if os.path.exists(os.path.join(BASE_DIR, "hc/local_settings.py")):
    from .local_settings import *
```
with a fixed but might be better, but again, we would need a pre-defined path for that. We could use something like /etc/healthchecks/settings.py for that. What do you think?

Reply:
@SuperSandro2000 I just implemented a solution using EnvironmentFile. The healthchecksManageScript wrapper became a bit more hacky and could probably be improved but it seems to work and configuration handling in general is much better than before IMO.